contracts/staking/fuse/temp/StakingRewardsPerEpoch.sol

```diff
@@ -0,0 +1,207 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "./StakingRewards.sol";
+import "./GoodDollarSwaps.sol";
+import "./IConsensus.sol";
+
+contract StakingRewardsPerEpoch is StakingRewards, GoodDollarSwaps {
+  using SafeERC20 for IERC20;
+
+  struct StakeInfoPerEpoch {
+    uint256 pendingStake;
+    uint256 giveBackRatio;
+    uint256 indexOfLastEpochStaked;
+  }
+
+  mapping (address => StakeInfoPerEpoch) public stakersInfoPerEpoch;
+
+  uint256 public constant RATIO_BASE = 10000;
+
+  address[] public validators;
+
+  IConsensus public consensus;
+
+  Uniswap public uniswapV2Router;
+  IGoodDollar public goodDollar;
+  IUBIScheme public ubiScheme;
+  UniswapFactory public uniswapFactory;
+  UniswapPair public uniswapGoodDollarFusePair;
+
+  uint256 public maxSlippageRatio; //actually its max price impact ratio
+
+  uint256 public keeperAndCommunityPoolRatio;
+  uint256 public communityPoolBalance;
```

Comment:
irrelevant. this is fusestaking stuff
you are writing a generic contract for rewards calculations only

Reply:
The code was removed from this generic.